velox/functions/prestosql/JsonFunctions.h

```diff
@@ -74,4 +74,39 @@ struct JsonArrayLengthFunction {
   }
 };
 
+template <typename T>
+struct JsonArrayContainsFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  template <typename TInput>
+  FOLLY_ALWAYS_INLINE bool
+  call(bool& result, const arg_type<Varchar>& json, const TInput& value) {
+    auto parsedJson = folly::parseJson(json);
+    if (!parsedJson.isArray()) {
+      return false;
+    }
+
+    result = false;
+    bool valueBool = std::is_same_v<TInput, bool>;
+    bool valueInt = std::is_same_v<TInput, int64_t>;
+    bool valueDouble = std::is_same_v<TInput, double>;
+    for (const auto& v : parsedJson) {
+      if (valueBool && v.isBool() && v == value) {
```

Comment:
`std::is_same_v<TInput, bool>` can be determined at compile time, so it's better not to do runtime checks on it. (Same for int64_t and double.) With the current code, valueBool, valueInt, and valueDouble are computed and checked at runtime. Consider using `if constexpr (...)` for type checks and do the value comparison inside. The rest looks good to me.

Reply:
Thanks this is a good suggestion. Agree its best to handle this in compilation. Otherwise choosing to write the templated code while taking these runtime hits is not worth the tradeoff.

Reply:
Thanks for the suggestion, changed accordingly